protocol/x/clob/keeper/orders.go

```diff
@@ -516,9 +554,30 @@ func (k Keeper) PerformOrderCancellationStatefulValidation(
 ) error {
 	orderIdToCancel := msgCancelOrder.GetOrderId()
 	if orderIdToCancel.IsStatefulOrder() {
-		cancelGoodTilBlockTime := msgCancelOrder.GetGoodTilBlockTime()
 		previousBlockTime := k.blockTimeKeeper.GetPreviousBlockInfo(ctx).Timestamp
 
+		prevBlockHeight := k.blockTimeKeeper.GetPreviousBlockInfo(ctx).Height
+		if lib.IsDeliverTxMode(ctx) && prevBlockHeight != uint32(ctx.BlockHeight()-1) {
```

Comment:
nit: I think we should also build a `GetBlockHeight(ctx)` function which returns height based on deliver tx/ check tx mode